std/testing/asserts.ts

```diff
@@ -19,10 +19,13 @@ export class AssertionError extends Error {
   }
 }
 
-function format(v: unknown): string {
-  let string = globalThis.Deno ? Deno.inspect(v) : String(v);
-  if (typeof v == "string") {
-    string = `"${string.replace(/(?=["\\])/g, "\\")}"`;
+/**
+ * Format value to string for output to CLI on error.
+ */
+function format(value: unknown): string {
+  const string = globalThis.Deno ? Deno.inspect(value) : String(value);
+  if (typeof value === "string") {
+    return `"${string}"`;
   }
   return string;
 }
```

Comment:
Doesn't really matter but I think the smaller refactors are unnecessary. The `string` variable being reassigned is good because it would uniquely refer to the return value and an early return isn't really a good thing if it doesn't save lines.

I _really_ don't see why people insist on using strict equality between `typeof x` and a static string...

Reply:
@nayeemrmn Everyone defaults to triple equals because we've been trained to. I would love to get JavaScript back to a place where we can use == with confidence. Hopefully with deno lint rules.

Reply:
I think it's wise to use strict equality for two reasons, first there is no reason not to and it's a good habit to have. Secondly this code handles testing and as such should be as robust as possible, even if it seems pedantic. Other modules which handle less sensitive areas can be more flexible in their approach.

In terms of the other points ideally I would rewrite the method in a similar way to this.
```js
function stringify(value: unknown) {
  return globalThis.Deno ? Deno.inspect(value) : String(value);
}

/**
 * Format value to string for output to CLI on error.
 */
function format(value: unknown): string {
  if (typeof value === "string") {
    return `"${stringify(value)}"`;
  }
  return stringify(value);
}
```
This code is cleaner, clearer, all the assignment stuff is gone and each method is only doing one thing. This has a number of benefits, especially given a lot of business logic is hidden here `globalThis.Deno ? Deno.inspect(value) : String(value);` . Also as testing private methods in JavaScript is hacky I'd put most of the message business logic in its own module so it can be properly tested in isolation. This together would result in greater code confidence in this sensitive area.

I haven't done this though because that would be unreasonable given my knowledge of the project, and I don't wish to tread on any toes, just help where I can.